<?xml version="1.0"?>
<launch>
  <arg name="input_clouds_not_mesh" default="true" />
  <arg name="mesh" default="" />
  <arg name="cloud1" default="" />
  <arg name="cloud2" default="" />
  <arg name="cloud3" default="" />
  <arg name="cloud4" default="" />
  <arg name="radius_normals" default="0.20" />
  <arg name="radius_robot" default="0.40" />
  <arg name="curvature_threshold" default="0.04" />
  <arg name="mesh_sampling_resolution" default="0.10" />

  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find gaitmesh_ros)/rviz/annotate_cloud_gaits.rviz" />

  <node pkg="gaitmesh_ros" type="mesh_from_mapping_server" name="mesh_from_mapping_server" output="screen"/>

  <node pkg="gaitmesh_ros" type="annotate_cloud_gaits" name="annotate_cloud_gaits" output="screen">
    <param name="radius_normals" value="$(arg radius_normals)" />
    <param name="radius_robot" value="$(arg radius_robot)" />
    <param name="curvature_threshold" value="$(arg curvature_threshold)" />
  </node>

  <node pkg="gaitmesh_ros" type="publish_initial_cloud" name="publish_from_file" output="screen">
    <param name="input_clouds_not_mesh" value="$(arg input_clouds_not_mesh)" />
    <param name="mesh_sampling_resolution" value="$(arg mesh_sampling_resolution)" />
    <param name="mesh" value="$(arg mesh)" />
    <param name="cloud1" value="$(arg cloud1)" />
    <param name="cloud2" value="$(arg cloud2)" />
    <param name="cloud3" value="$(arg cloud3)" />
    <param name="cloud4" value="$(arg cloud4)" />
    <param name="reference_point_x" value="5" />
    <param name="reference_point_y" value="-3" />
    <param name="reference_point_z" value="0.8" />
  </node>
</launch>

    <!-- example model from Barcelona Robot Lab Dataset -->
<!--
    <param name="mesh" value="$(find gaitmesh_ros)/data/BRL_final.obj" />
-->
    <!-- oil_rig model -->
<!--
    <param name="mesh" value="$(find recast_demos)/data/oilRig_detailed.obj" />
-->
    <!-- Blyth #ORCAHubDemos -->
<!--
    <param name="cloud1" value="/home/martim/ORCAHubDemos/blyth-01-10-2019-5cm-downsampled-cleaned2-cutceiling-cutside-nopillars-cleanfloor.ply" />
    <param name="cloud_mesh" value="/home/martim/ORCAHubDemos/mesh-blyth-01-10-2019-5cm-downsampled-cleaned2-cutceiling-cutside-nopillars-cleanfloor.obj" />
-->
    <!-- FSC full model -->
<!--
    <param name="cloud1" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-merged-1stfloor-1cm-clipped-icp.ply" />
    <param name="cloud2" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/mesh-oilrig-merged-main-1cm-clipped-icp-clean.ply" />
    <param name="cloud3" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-merged-west-1cm-clipped-icp.ply" />
    <param name="cloud_mesh" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/mesh-oilrig-merged-all.obj" />
-->
    <!-- FSC main floor only -->
<!--
    <param name="cloud1" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-merged-main-1cm-clipped-icp-clean3.ply" />
    <param name="cloud_mesh" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/mesh-oilrig-merged-main-1cm-clipped-icp-clean3.obj" />
-->
    <!-- FSC container1 only -->
<!--
    <param name="cloud2" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-container-merged-1cm-aligned-normals-ascii-filtered.ply" />
    <param name="cloud_mesh" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/mesh-oilrig-container-merged-1cm-aligned-normals-ascii-filtered.obj" />
-->
    <!-- FSC container1 and main -->
<!--
    <param name="cloud1" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-merged-main-1cm-clipped-icp-clean3.ply" />
    <param name="cloud2" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-container-merged-1cm-aligned-normals-ascii-filtered-realign-cutmain.ply" />
    <param name="cloud_mesh" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/mesh-oilrig-container-main-merged.obj" />
-->
    <!-- FSC container1 and full model -->
<!--
    <param name="cloud1" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-merged-1stfloor-1cm-clipped-icp.ply" />
    <param name="cloud2" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-merged-main-1cm-clipped-icp-clean3.ply" />
    <param name="cloud3" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-merged-west-1cm-clipped-icp-clean.ply" />
    <param name="cloud4" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-container-merged-1cm-aligned-normals-ascii-filtered-realign-cutmain.ply" />
    <param name="cloud_mesh" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/mesh-oilrig-final-main-container-1stfloor-west.obj" />
-->
    <!-- ORI map -->
<!--
    <param name="cloud1" value="/home/martim/drs_testing_data/leica/ori-building-maps/map_filter_cutaway_for_autonomy_5cm.ply" />
    <param name="cloud_mesh" value="/home/martim/drs_testing_data/leica/ori-building-maps/mesh_holefilled_map_filter_cutaway_for_autonomy_5cm.obj" />
-->
